### DISCUSSION TITLE
Invalid time value: "There was a problem loading your grid" with version 2.27.3

### DISCUSSION THREAD

**Discussion by Guenter9** [Budibase/budibase#13798] [Help]
Since I update my app to version 2.27.3. I've got problems with my tables containing a timestamp type of column (using a SQLServer as datasource). I tried to revert the to the former version, but the issue still remains. 
Displaying the table in the data section leads to the message "There was a problem loading your grid - Invalid time value" 
The preview of the screen is also frequenly showing the message " Invalid time value" 
I sure need the timetamp field in my tables for some reason. So the app is now ruined and no way out? 
Any help would be appreciated.

**Comment by mikesealey** [comment]
Hi @Guenter9 

I've seen this once before - the root cause was that Budibase was expecting a Date & Time in the column, and it was only being passed a Date without the time (thus "throwing the invalid time value" error). Our engineering team are aware of the issue, and looking into it.

This is a recoverable position to be in, _please don't worry_. If your tables are still working/displaying in the GUI you might be able to create a table and form to update the row with the invalid time value.

You could also create a new query in the Data tab by clicking on your data source, and going to the Queries subheading.
![image](https://github.com/Budibase/budibase/assets/39269380/547c7058-9f85-4a22-992c-a3974ae09a98)


There you could create a READ query to select `row_id` and `date_time` from your table where date_time <is bad, null, not as expected, etc>. Make a note of the `row_id` on affected rows, and write a new query to set them to a valid date_time.

Please could you also share the schema used in the affected tables?

Thanks
Mike

**Reply by Guenter9** [reply]
Hi Mike,
sorry for the delayed answer. The problem still is arising from using a special data type of SQL-Server

Using the datatype "timestamp" in a table in SQLServer will cause the problem, since this datatype contains not a timestamp in common (datetime Type) sense. Instead, it is a rather a row-version (I guess in some hexadecimal format). 
Since the datatype is meanwhile is depreciated, I tried to use the recommended Type "rowversion", which is some sort of binary type.
The data type works well in my main frontend application. In Budibase I encounter the same problem loading the grid. So same as with timestamp. So my conclusion is, that the interpretation of data types and their interpretation in Budibase must have changed. Formerly, the timestap-type didn' t cause a problem.

Here is the SQL for testing a table with this kind of datatype:
```sql
CREATE TABLE [dbo].[tbl_test](
	[ID] [int] IDENTITY(1,1) NOT NULL,
	[Kennung] [nvarchar](255) NULL,
	[Titel] [nvarchar](30) NULL,
	[RowVersion] [rowversion] NOT NULL
	CONSTRAINT [tbl_test$PrimaryKey] PRIMARY KEY CLUSTERED 
(
	[ID] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
) ON [PRIMARY]
```
Fetching and displaying the table will lead to the error "There was a problem loading your grid
Invalid time value"

**Reply by dado-ca** [reply]
@Guenter9  Hi, were you able to solve this issue? I have the same issue, with date fields after updating to newer version of Budibase. I am using Postgres database.

**Comment by Guenter9** [comment]
Actually no. Finally I omitted the use of the timestamp types in the database. Am 13.08.2024 um 15:51 schrieb Damir Merdan ***@***.***>:﻿
@Guenter9  Hi, were you able to solve this issue? I have the same issue, with date fields after updating to newer version of Budibase. I am using Postgres database.
